cli/src/pcluster/templates/imagebuilder_stack.py

```diff
@@ -56,53 +74,30 @@ def __init__(self, scope: core.Construct, construct_id: str, imagebuild: ImageBu
         # Setup ImageBuilder Resources
         resources_prefix = utils.generate_random_prefix()
 
-        # InstanceRole
-        managed_policy_arns = [
-            core.Fn.sub("arn:${AWS::Partition}:iam::aws:policy/AmazonSSMManagedInstanceCore"),
-            core.Fn.sub("arn:${AWS::Partition}:iam::aws:policy/EC2InstanceProfileForImageBuilder"),
-        ]
+        # InstanceRole and InstanceProfile
+        instance_profile_name = None
         if build.instance_role:
-            managed_policy_arns.extend([build.instance_role])
-
-        instancerole = iam.CfnRole(
-            self,
-            id="InstanceRole",
-            managed_policy_arns=managed_policy_arns,
-            assume_role_policy_document={
-                "Statement": {
-                    "Action": "sts:AssumeRole",
-                    "Effect": "Allow",
-                    "Principal": {"Service": "ec2.amazonaws.com"},
-                },
-                "Version": "2012-10-17",
-            },
-            path="/executionServiceEC2Role/",
-        )
-        instancerole.add_metadata("Comment", "Role to be used by instance during image build.")
-
-        # InstanceProfile
-        iam.CfnInstanceProfile(
-            self, id="InstanceProfile", path="/executionServiceEC2Role/", roles=[core.Fn.ref("InstanceRole")]
-        )
+            instance_role_type = self._get_instance_role_type()
+            if instance_role_type == InstanceRole.ROLE:
+                self._set_instance_profile(instance_role=build.instance_role)
+            elif instance_role_type == InstanceRole.INSTANCE_PROFILE:
+                instance_profile_name = build.instance_role
+            else:
+                self._set_instance_profile()
```

Comment:
what is this `else` for?

Reply:
I was thinking if there exists other cases, but it is useless since we validate arn in schema layer, removed.